etc/notebooks/got_scotch_demo/scotch_dashboard_declarative.ipynb

```diff
@@ -0,0 +1,783 @@
+    "                <paper-item>{{scotch}}</paper-item>\n",
+    "            </template>\n",
+    "        </paper-menu>\n",
+    "    </paper-dropdown-menu>\n",
+    "</template>\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "urth": {
+     "dashboard": {
+      "hidden": true
+     }
+    }
+   },
+   "source": [
+    "### Suggestion Table"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "urth": {
+     "dashboard": {
+      "hidden": true
+     }
+    }
+   },
+   "source": [
+    "Now let's add an urth-viz-table to display the 5 similar scotches to the one that user picks from the above dropdown.\n",
+    "\n",
+    "The data this table needs will be the result from calling our function *get_similar_for_table*. So we will use an urth-core-function to call this function with arguments *name* and *n*, with *name* being the selected Scotch, and *n* is 6 (to retrieve top 5 similar scotches and itself). The return of this function is a dataframe that urth-viz-table will then make use of, we will call it *suggestedTbl*.\n",
+    "\n",
+    "User has the ability to click any scotch from this table to compare further the taste profile of this scotch compared to the one from the dropdown. We will store user selection to a variable called *similarScotch* so we can use it to generate our feature keys list later."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": false,
+    "urth": {
+     "dashboard": {
+      "layout": {
+       "col": 4,
+       "height": 6,
+       "row": 4,
+       "width": 8
+      }
+     }
+    }
+   },
+   "outputs": [],
+   "source": [
+    "%%html\n",
+    "<template is=\"urth-core-bind\">\n",
+    "    <urth-core-function id=\"getSimilarFunc\" ref=\"get_similar_for_table\" arg-name=\"{{selectedScotch}}\" arg-n=\"6\" result=\"{{suggestedTbl}}\" auto></urth-core-function>\n",
+    "    <urth-viz-table id=\"suggestionTable\" datarows='{{suggestedTbl.data}}' columns='{{suggestedTbl.columns}}' selection=\"{{similarScotch}}\"></urth-viz-table>\n",
+    "</template>"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "urth": {
+     "dashboard": {
+      "hidden": true
+     }
+    }
+   },
+   "source": [
+    "### Feature Chart"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "urth": {
+     "dashboard": {
+      "hidden": true
+     }
+    }
+   },
+   "source": [
+    "Lastly, we will display the scotches features with a variety of urth-viz-* elements.\n",
+    "\n",
+    "We need to use urth-core-function to call *get_feature_keys*, give it the selected scotch name, the similar scotch name user clicked on, and the result of the function call is a list of these two names. We will assign the result to a variable called *feature_keys* that another function *get_features_for_chart* will make use of right after this."
```

Comment:
OK. Here's where that mystery function comes into play. I'm starting to understand that it's used to build a tuple that'll be used to track the drop down selected scotch and the last clicked scotch.